dasharo-compatibility/power-after-fail.robot

```diff
@@ -0,0 +1,94 @@
+*** Settings ***
+Library             Collections
+Library             OperatingSystem
+Library             Process
+Library             String
+Library             Telnet    timeout=20 seconds    connection_timeout=120 seconds
+Library             SSHLibrary    timeout=90 seconds
+Library             RequestsLibrary
+# TODO: maybe have a single file to include if we need to include the same
+# stuff in all test cases
+Resource            ../sonoff-rest-api/sonoff-api.robot
+Resource            ../rtectrl-rest-api/rtectrl.robot
+Resource            ../variables.robot
+Resource            ../keywords.robot
+Resource            ../keys.robot
+Resource            ../lib/bios/menus.robot
+
+# TODO:
+# - document which setup/teardown keywords to use and what are they doing
+# - go threough them and make sure they are doing what the name suggest (not
+# exactly the case right now)
+Suite Setup         Run Keyword
+...                     Prepare Test Suite
+Suite Teardown      Run Keyword
+...                     Log Out And Close Connection
+
+
+*** Test Cases ***
+PSF001.001 Check Power State After Power Failure default state (firmware)
+    [Documentation]    This test ensures that the option is present, and the
+    ...    default state of this option after flashing is correct.
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    PSF001.001 not supported
+    Power On
+    Enter Dasharo System Features
+    Enter Power Management Options
+    ${psf_state}=    Read From Terminal Until    ESC to exit.
+    Should Contain    ${psf_state}    Powered Off
+
+PSF002.001 Powered Off State Restoration Test
+    [Documentation]    This test ensures that the feature is able to
+    ...    keep the DUT powered off after power failure.
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    PSF002.001 not supported
+    Power On
+    Enter Dasharo System Features
+    Enter Power Management Options
+    Set Power After Fail State    Powered Off
+    Save Changes And Reset    2    4
+    Simulate Power Failure
+    Sleep    5s
+    ${output}=    Read From Terminal
+    Length Should Be    ${output}    1
```

Comment:
Why do we sleep?

We should just wait for the prompt from the firmware using `Read From Terminal Until    <prompt>` or `Enter Setup Menu Tianocore`  and expect failure/timeout (because nothing should be printed as machine should stay powered off).

Reply:
Fixed in [recent commit](https://github.com/Dasharo/open-source-firmware-validation/commit/e84e768780bae0482929f7309386c111a401257c)